docs/ilm.asciidoc

```diff
@@ -2,158 +2,309 @@
+[source,yml]
+----
+apm-server:
+  ilm:
+    setup:
+      enabled: true
+      overwrite: true
+      require_policy: true
+      templates:
+        - event_type: "error"
+          policy_name: "apm-rollover-30-days"
+        - event_type: "span"
+          policy_name: "apm-rollover-30-days"
+        - event_type: "transaction"
+          policy_name: "apm-rollover-30-days"
+        - event_type: "metric"
+          policy_name: "apm-rollover-30-days"
+----
 
 [float]
-===== `ilm.enabled: false`
+[[ilm-setup-config]]
+===== `apm-server.ilm.setup.enabled`
+
+Defaults to `true`.
 
-ILM can be explicitly disabled by setting `ilm.enabled` to `false`.
+When `true`, APM Server will create an ILM specific index template for each APM Server event type.
+This is required to map ILM aliases and policies to indices.
+
+When `false`, ILM setup is disabled.No policies, templates, or aliases will be created by APM Server.
+Only disable `ilm.setup` if you want to set up index management on your own.
+If you simply want to disable ILM, use `apm-server.ilm.enabled: false` instead.
 
 [float]
-[[ilm-setup]]
-==== ILM default policy setup
+[[ilm-setup-overwrite-config]]
+===== `apm-server.ilm.setup.overwrite`
 
-It is recommended to set up index lifecycle management (ILM) before starting APM Server.
-This excludes setup from the ingestion process, which allows you to ensure ILM is set up correctly before using APM.
+Defaults to `false`. When `false`, APM Server will not overwrite any existing policies or ILM related templates.
+When first setting up ILM, your initial template and policy will be applied.
+You must set this to `true` when customizing your policies and template for them to be applied.
 
-Run the <<setup-command>> with the ` --index-management` flag to set up the default ILM policy:
+[float]
+[[ilm-setup-policy-config]]
+===== `apm-server.ilm.setup.require_policy`
 
-[source,js]
------------------------
-./apm-server setup --index-management
------------------------
-// CONSOLE
+Defaults to `true`, which means that an ILM policy must be defined in `apm-server.yml`.
+Changing this to `false` allows you to manually set up ILM policies and templates outside of APM Server,
+e.g., with Elasticsearch APIs.
+APM Server will still make use of ILM and connect your template with the defined mapping.
 
-You can confirm the policy was created with the GET lifecycle policy API.
-Here's what the transaction response looks like:
+[float]
+[[ilm-setup-mapping-config]]
+===== `apm-server.ilm.setup.mapping`
+
+Maps each event type to the named policy. APM Server event types can only be `error`, `span`, `transaction`, and `metric`.
+If you attempt to map an index lifecycle policy to a different event type,
+or one of the event types is missing form this mapping, APM Server will not start.
+
+Policies defined must be mapped to an event type. If they are not, they will not be sent to Elasticsearch.
+
+[float]
+[[custom-ilm-policy]]
+==== Create a custom ILM policy
```

Comment:
It might also be worth explaining that any changes in existing ILM policies that are already applied to indices only take place when the next phase is entered. There's a detailed explanation at [ES update-policy](https://www.elastic.co/guide/en/elasticsearch/reference/current/update-lifecycle-policy.html)